huaweicloud/services/live/resource_huaweicloud_live_record_callback.go

```diff
@@ -0,0 +1,228 @@
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Live callback configuration")
+	}
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("domain_name", response.PublishDomain),
+		setTypeToState(d, response.NotifyEventSubscription),
+		d.Set("url", response.NotifyCallbackUrl),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceRecordCallbackUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcLiveV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Live v1 client: %s", err)
+	}
+
+	updateBody, err := buildCallbackConfigParams(d, region)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	updateOpts := &model.UpdateRecordCallbackConfigRequest{
+		Id:   d.Id(),
+		Body: updateBody,
+	}
+
+	_, err = client.UpdateRecordCallbackConfig(updateOpts)
+	if err != nil {
+		return diag.Errorf("error updating Live callback configuration: %s", err)
+	}
+
+	return resourceRecordCallbackRead(ctx, d, meta)
+}
+
+func resourceRecordCallbackDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcLiveV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Live v1 client: %s", err)
+	}
+
+	deleteOpts := &model.DeleteRecordCallbackConfigRequest{
+		Id: d.Id(),
+	}
+	_, err = client.DeleteRecordCallbackConfig(deleteOpts)
+	if err != nil {
+		return diag.Errorf("error deleting Live callback configuration: %s", err)
+	}
+
+	return nil
+}
+
+func buildCallbackConfigParams(d *schema.ResourceData, region string) (*model.RecordCallbackConfigRequest, error) {
+	v := d.Get("type").([]interface{})
+	events := make([]model.RecordCallbackConfigRequestNotifyEventSubscription, len(v))
+	for i, v := range v {
+		var event model.RecordCallbackConfigRequestNotifyEventSubscription
+		err := event.UnmarshalJSON([]byte(v.(string)))
+		if err != nil {
+			return nil, fmt.Errorf("error getting the argument %q: %s", "type", err)
+		}
+		events[i] = event
+	}
+
+	req := model.RecordCallbackConfigRequest{
+		PublishDomain:           d.Get("domain_name").(string),
+		App:                     allAppName,
+		NotifyEventSubscription: &events,
+		NotifyCallbackUrl:       utils.String(d.Get("url").(string)),
+	}
+
+	return &req, nil
+
+}
+
+func getRecordCallBackId(client *v1.LiveClient, publishDomain, appName string) (string, error) {
+	resp, err := client.ListRecordCallbackConfigs(&model.ListRecordCallbackConfigsRequest{
+		PublishDomain: &publishDomain,
+		App:           &appName,
+	})
+	if err != nil {
+		return "", fmt.Errorf("error retrieving Live callback configuration: %s", err)
+	}
+
+	if resp == nil || resp.CallbackConfig == nil || len(*resp.CallbackConfig) < 1 {
+		return "", fmt.Errorf("error retrieving Live callback configuration: %s", err)
```

Comment:
this error will always be `<nil>`, pls remove it and update the description.

Reply:
done